File: android/lib/src/main/java/com/velocitysystems/downloadmanager/DownloadStatus.kt
package com.velocitysystems.downloadmanager

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Represents the various states of a download item.
 */
@Serializable
enum class DownloadStatus {
    /// Status could not be determined.
    @SerialName("unknown")
    Unknown,

    /// Download has not yet been created/persisted.
    @SerialName("pending")
    Pending,

    /// Download has been created and is ready to start.
    @SerialName("idle")
    Idle,

    /// Download is in progress.
    @SerialName("inProgress")
    InProgress,

    /// Download was in progress but has been paused.
    @SerialName("paused")
    Paused,

    /// Download was cancelled by the user.
    @SerialName("cancelled")
    Cancelled,

    /// Download completed.
    @SerialName("completed")
    Completed,
}

File: ios/DownloadManagerKit/Sources/DownloadManagerKit/DownloadStatus.swift
//
//  DownloadStatus.swift
//  DownloadManagerKit
//

/// Represents the various states of a download item.
public enum DownloadStatus: String, Codable, CaseIterable, Sendable {
   /// Download has been created and is ready to start.
   case idle
   /// Download is in progress.
   case inProgress
   /// Download was in progress but has been paused.
   case paused
   /// Transfer failed; resume retries it and cancel discards it.
   case failed
   /// Download was canceled by the user.
   case canceled
   /// Download completed.
   case completed
}
